src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidUnusedPrivateFields.cs

```diff
@@ -48,12 +48,27 @@ public override void Initialize(AnalysisContext analysisContext)
                     HashSet<IFieldSymbol> unreferencedPrivateFields = new HashSet<IFieldSymbol>();
                     HashSet<IFieldSymbol> referencedPrivateFields = new HashSet<IFieldSymbol>();
 
+                    // Fields with certain special attributes should never be considered unused.
+                    ImmutableHashSet<INamedTypeSymbol> specialAttributes = GetSpecialAttributes(compilationContext.Compilation);
+                    
                     compilationContext.RegisterSymbolAction(
                         (symbolContext) =>
                         {
                             IFieldSymbol field = (IFieldSymbol)symbolContext.Symbol;
                             if (field.DeclaredAccessibility == Accessibility.Private && !referencedPrivateFields.Contains(field))
                             {
+                                if (!specialAttributes.IsEmpty)
+                                {
+                                    var fieldAttributes = field.GetAttributes();
```

Comment:
Unused?

Reply:
Yes, how ironic :-) We should really enable analyzers again on our own repo.